client/state/plugins/installed/actions.js

```diff
@@ -532,3 +535,32 @@ export function fetchSitePlugins( siteId ) {
 export function fetchPlugins( siteIds ) {
 	return ( dispatch ) => siteIds.map( ( siteId ) => dispatch( fetchSitePlugins( siteId ) ) );
 }
+
+export function fetchAllPlugins() {
+	return ( dispatch ) => {
+		dispatch( { type: ALL_PLUGINS_REQUEST } );
+
+		const receivePluginsDispatchSuccess = ( { sites } ) => {
+			dispatch( { type: ALL_PLUGINS_REQUEST_SUCCESS } );
+
+			Object.entries( sites ).map( ( [ siteId, plugins = [] ] ) => {
+				dispatch( receiveSitePlugins( siteId, plugins ) );
+
+				plugins.map( ( plugin ) => {
+					if ( plugin.update && plugin.autoupdate ) {
+						updatePlugin( siteId, plugin )( dispatch );
+					}
+				} );
+			} );
+		};
+
+		const receivePluginsDispatchFail = ( error ) => {
+			dispatch( { type: ALL_PLUGINS_REQUEST_FAILURE, error } );
+		};
+
+		return wpcom.req
+			.get( `/me/sites/plugins` )
+			.then( receivePluginsDispatchSuccess )
+			.catch( receivePluginsDispatchFail );
```

Comment:
What kind of error handling are we doing, just swallowing the errors?

Reply:
Yes, it looks like the `PLUGINS_REQUEST_FAILURE` is passing the actual message through, so I did the same for the new endpoint, but I created a new card in Asana to fix this.

1202518759611394-as-1202789398406899